### DISCUSSION TITLE
Possible new maps to add

### DISCUSSION THREAD

**Discussion by AbusedPet** [DarthVan/StarCraft2-Bank-Restorer#2] [Ideas]
Your work is pretty amazing the second I found this project I nutted instantly on the spot. You seem to be making frequent updates and I thought I could suggest some maps to be added if it's not too noisy for me:
1. Undead assault reborn 
2. Star battle reloaded 
3. Kerrigan survival 
4. Trancespace Fortress 3
5. Haven Forces
6. Night of the Dead
7. The thing
8. Direct Strike (pretty much just the skins without premium obviously that we can unlock and xp? I'm adding it just because I'm like 1k level and just want a max 3k account :D)
Pretty much some of the most played maps that are encrypted and protected 
Cheers for your work I hope you continue working on this it's pretty dope

**Comment by SquogSamantha** [comment]
Fallen World Last Stand
Pandorium Survivors

**Comment by DarthVan** [comment]
@AbusedPet, @SquogSamantha, thanks!

I'm a lot busy at the moment, so I can't guarantee that there will be patches anytime soon. **Minerals Evolution** is next in line, there is a lot of farm there. after that I can take a new map from your list.

Also, if you have experience in map hacks, you can add a pull request here for the desired map

**Reply by AbusedPet** [reply]
Ayy so glad you're back and haven't abandoned this take all your sweet time in making this project bigger and better. 
I used to crack maps a long time ago and have since forgotten or grown stupider, so I can't help with encrypted ones.
I can't wait for your new update whenever it hits. All the good luck!

**Comment by Altogolik** [comment]
Trancespace Fortress 3
please. It's on the guy's list above.

**Reply by DarthVan** [reply]
I played this map only one time before and didnt understand what to do there :)
For hacking it i need an explanation, what params/stats/achives must be editable in the map form. I think, an article with screenshots and explanations will be enough to understand what parameters to include in the form.

p.s. did you tried StarBank for this?

**Reply by Altogolik** [reply]
Оно не работает, это довольно сложная карта.
Разработчик продает предметы по 40- 50 баксов
https://www.flipsnack.com/trancespacerewards/trancespace-fortress-3-rewards.html
Как я понял они все прописаны в банке и их можно активировать при желании.
+ в игре очень важен опыт (там можно хоть 40 тысяч уровней накопить)
и хероик поинты (для прокачки нужны)
Самое главное что хотелось бы, в игре есть предметы по типу Ruby of prestige. Чтобы его получить нужно вкачать 40000 уровень и обнулить персонажа. Второй раз начинать все сначала ну такое..
Это самый важный предмет)
https://docs.google.com/spreadsheets/d/1s-YNbr8gvRt_zhyTfidYD34fLShwkfKMMBCA274he34/edit?pli=1#gid=1985641173 (список всех предметов, самые ценные это сикрет итемс и улучшаемые предметы)

**Reply by Altogolik** [reply]
Плюс в игре есть статистика https://gyazo.com/627a0933197f80370866dfed15428874 (скрин)
Победы \ проигрыши \ убийства 
Валор это испытия которые нужно на героях проходить (их впринципе можно сделать самому не сложно) они для каждого героя отдельно проходятся и тут только общее число отображается пройденых.

Еще в карте есть ачивки, но их не проблема самому сделать, хотя верхние закрытые довольно сложные могут быть особенно пройти анрил соло (https://gyazo.com/0af0fc795500ab13e37356f8d1ada296)

Самое главное это секретные предметы (особенно рубин пристижа), валор поинты и опыт для прокачки. +статистика чтобы соотвествовать

**Reply by DarthVan** [reply]
Ok, thanks!

Similar mechanics were in **Zombie World: Liberty**. There was also an inventory where the jewels were stored, which give boosts to each player on the team. They can be obtained from bosses or at the end of the match, and the better item, the less likely its to get it.
The hack allows you to get any item in any quantity. 

It will probably be about the same here.

**Reply by Altogolik** [reply]
Надеюсь все получится) это первая карта где я вижу прямо такой явный донат с прайслистом) куча шмота с шансом выпадения 0.001%

**Reply by Altogolik** [reply]
Еще вспомнил, на карте есть разные герои их всех легко открыть кроме 4х секретных, их также надо выбивать с мизерным шансом, неплохо бы их тоже добавить в таблицу (скриншот прикладываю, закрытые герои секретные https://gyazo.com/bdddc6e74a949d7b961cd860022fbab4)

**Reply by Altogolik** [reply]
Есть успехи ? Если нужно что по карте обьяснить могу подсказать, мой вк такойже как тут ник

**Reply by DarthVan** [reply]
Патч 1.0.9 залил в выходные, но там фиксы по минералз и разным мелочам. А эту карту еще даже не начинал, завал на проекте)
Если что, Sith#2913 в батле добавь, я там на разные вопросы отвечаю, чтоб тут не флудить.

**Comment by SquogSamantha** [comment]
Seeing MineralZ added, I might suggest another one
MineralZ and the Thing rebalance
Also, which MineralZ Evolution game is added specifically?

**Reply by DarthVan** [reply]
The next map in development is **Trancespace Fortress 3**, cuz it has crazy grind and requires donations. People have asked for this several times.

> Also, which MineralZ Evolution game is added specifically?
- **MineralZ Evolution [Updated]**

Older version of mineralz ppl almost dont play.

**Reply by SquogSamantha** [reply]
Alright, thank you!

**Reply by Wolverinex01** [reply]
Any update on Trancespsace Fortress 3?

**Reply by DarthVan** [reply]
> Any update on Trancespsace Fortress 3?

Nope :(
Have no free time for this these months..

**Comment by Umokazan** [comment]
Do you have a discord to get in contact? Would like to get better and help with specific maps

**Reply by DarthVan** [reply]
Nop, for any questions **Sith#2913** in BattleNet.

**Comment by Ainzkrz** [comment]
thanx vm for this generator i can restore some bank for acarde    but someone have an idée for fallen world last stand i can't restore this arcade bank because I don't understand anything about this bank    ( sorry for my langage i little speak english ^^ )

**Reply by DarthVan** [reply]
Hi! Sorry guys, no free time at the moment, 146% busy on commercial projects >.<
They already asked for fallen world last stand, but i still can't finish current update..

**Comment by luxdercito** [comment]
plz add moonlords revamped

**Reply by DarthVan** [reply]
I'll try, if i get some free time) Still very busy..

**Comment by maks-yasnov** [comment]
Minerals survival classic
Although we wrote above that they almost don't play, but I would really like to see a hack (and in the evening you can usually find people who play)
Link to the map
battlenet:://starcraft/map/2/216446
The problem arises with this
Key name="CheatProtection2"

**Reply by DarthVan** [reply]
Ok, will try later.
Think that key is a checksum or something else.

**Reply by maks-yasnov** [reply]
Thanks in advance

**Reply by maks-yasnov** [reply]
StarBank is not working 
![cheat](https://github.com/user-attachments/assets/45f94f59-2910-4984-8c8d-ff17b67578ed)

![image](https://github.com/user-attachments/assets/40de1058-68f9-4c31-89a7-ac30f03ca3a8)

**Reply by DarthVan** [reply]
Starbank is a BlueRaja project, I have nothing to do with it. If you find an issue there, then create an issue in his repository - [StarBank issues](https://github.com/BlueRaja/Starbank/issues)

But it seems that he hasn't updated it for a long time. Now in the Galaxy code, a feature of saving keys from unit properties has begun to be used. This problem was raised both here and on his repository, and it is still not solved there.

Сейчас там кстати последний коммит 2х летней давности, и 2 непофикшенные проблемы с сигнатурой - не игнорируются ключи типа "Text" и нет поддержки составных ключей, как выше описано. Так что некоторые банки он сейчас просто не сможет корректно пересчитать, например от ZW:Reign of Chaos

**Reply by maks-yasnov** [reply]
I am waiting for your solution to this problem then

**Reply by DarthVan** [reply]
Try :)

![3](https://github.com/user-attachments/assets/c69f404b-fd1e-4b21-b6e9-c4f67b870a64)

It seems, i forgot about "cus" section with skin, color and other custom settings, but it should not affect the statistics 🤔

![3](https://github.com/user-attachments/assets/f20b51bb-1f76-46e1-97c6-14cc077f371c)

Don't forget about PlayerID. There are 2 anti-cheat keys on this map that use it.

**Reply by maks-yasnov** [reply]
For some reason it doesn't work for me, it says that cheats have been detected. I tried to change not only the last save, but also the last 3, it didn't help, after that I tried to set the maximum possible values, it didn't help either. I also tried to set the same result that I had in an honest way. It didn't work out.
I found a few inconsistencies: 

1. If you set Hell = 0 then 
Key name="NightsHell"
			Value int="1000"/

2. Experience has a non-integer value:

Key name="Experience"
            Value fixed="27773.8151"/

3. At the end of your file there is this line:
Signature value="BA99914ABA1BFDFB395AE01587AB87908B9BD8F2"/

Although, there is no such thing in a normal file
the values that I played without cheats do not work:
![image](https://github.com/user-attachments/assets/9904bfb4-83f2-4743-aee6-332789f407fb)

I can't understand why you did well and I didn't.

**Reply by maks-yasnov** [reply]
And another question. Maybe I'm translating all this through a translator in vain, because it feels like we're from the same country. :)
At the bottom of the photo it says:
" легк. - механич. - особ. "
)))

**Reply by DarthVan** [reply]
Так ведь по нику и так ясно, что same кантри)
1. Мне кажется, ты неправильно вводишь **Player ID**. Это самая частая причина у кого не работает.
2. Если тестируешь в регионе EU, то **Author ID** по умолчанию прописан корректный. Однако для других регионов этот параметр так же надо сверять с путем к файлу, и заменить значение при необходимости (в Help с рисунками об этом расписано)
3. Если 1. и 2. не помогло, то надо разбираться, но у меня все завелось с первого раза.

ПС: то что есть сигнатура в конце файла, роли не играет, просто карта не использует ее. Вместо сигнатуры валидность проверяется по двум античит ключам.

**Reply by maks-yasnov** [reply]
уже несколько раз проверил, но скорее всего что-то другое

**Reply by maks-yasnov** [reply]
![image](https://github.com/user-attachments/assets/d83912a0-e235-47ae-9071-6aaecf15d618)

**Reply by DarthVan** [reply]
Странно, вроде все должно быть ок. Проверю потом. Не пробовал сгенерировать банк без загрузки своих сейвов? нажать там Reset и посмотреть, загрузит или нет

**Reply by maks-yasnov** [reply]
Проверил, не работает.
Также убрал 
Signature value="BA99914ABA1BFDFB395AE01587AB87908B9BD8F2" 
Не помогло

Если не сложно скинь файл который ты сделал, я посмотрю и попробую те же значения сделать

**Reply by DarthVan** [reply]
Да, проверил щас, какие-то проблемы есть, через раз банки обнуляются, возможно какое-то условие не выполняется.

Попробуй пока этот банк [MSC bank.zip](https://github.com/user-attachments/files/17089284/MSC.bank.zip) (уже сгенерен под твой Player ID), на моей учетке эти статы заработали, когда nights чуть изменил, в порядке убывания:

![4](https://github.com/user-attachments/assets/9151c270-21c9-45ad-b9fd-311f24c26b85)

Щас пока время нет возиться, но потом надо будет скрипт проверить от карты, за что там еще reset делают, если хочешь, заведи пока issue.


Вот такие условия сейчас есть:
```
if (((lib1_gv_version != BankValueGetAsFixed(BankLastCreated(), "sec", "version")) ||
    (BankValueGetAsFixed(BankLastCreated(), "sec", "CheatProtection1") != BankValueGetAsFixed(BankLastCreated(), "sec", "CheatProtection3")) ||
    (BankValueGetAsFixed(BankLastCreated(), "sec", "CheatProtection2") != BankValueGetAsFixed(BankLastCreated(), "sec", "CheatProtection4")) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "NightsEasy") < BankValueGetAsInt(BankLastCreated(), "sec", "NightsNormal")) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "NightsNormal") < BankValueGetAsInt(BankLastCreated(), "sec", "NightsHard")) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "NightsHard") < BankValueGetAsInt(BankLastCreated(), "sec", "NightsInsane")) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "NightsInsane") < BankValueGetAsInt(BankLastCreated(), "sec", "NightsHell")) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "NightsEasy") > 100) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "NightsNormal") > 100) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "NightsHard") > 100) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "NightsInsane") > 100) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "NightsHell") > 100) ||
    (BankValueGetAsInt(BankLastCreated(), "sec", "Wins") > BankValueGetAsInt(BankLastCreated(), "sec", "Games")))) {
    
    // читер
    lib1_gv_enough[lp_player] = false;
			
```
Видимо, мне надо еще проверить js имплементацию мат.функций старика :)

**Reply by maks-yasnov** [reply]
![image](https://github.com/user-attachments/assets/6091f862-8212-4572-aaf3-a93837e61f1b)

**Reply by DarthVan** [reply]
не везет)

**Reply by SquogSamantha** [reply]
me casually using deepL to translate this conversation lol

**Reply by DarthVan** [reply]
Have a same problem?)

I will try to fix this later, seems it need some more time.

**Reply by SquogSamantha** [reply]
yeah i actually tried the same and got the same problem

**Reply by DarthVan** [reply]
Ok, will try to fix this.

**Comment by SquogSamantha** [comment]
zombie world liberty is broken on the website btw
whenever I open it the web screen just goes completely black, had to reset cookies to use the web again

**Reply by DarthVan** [reply]
ups, it's probably because of the map position shift, because there are more maps in front of it now. Thanks, I'll check it soon.
PS. clearing LocalSotrage's variables for the web site must be enough to fix this.

**Reply by DarthVan** [reply]
Fixed.

**Reply by SquogSamantha** [reply]
yipee! thank you

**Comment by LamethDaveth** [comment]
Can you add the eternal wrath of eres test version? My stats got shit on and I can't get it back. Farming in that one is a nightmare too.

Love that it works for zombie world unity though, got all my stats back after my original back got deleted.

PS: i think the zombie world liberty dev added something for that game that made changes detectable. Not sure though.

**Reply by DarthVan** [reply]
Not sure i have enough free time, but ill check.

If you can't generate a valid ZW:Liberty / Unity bank file, please post a bug to one of issues.

**Comment by Capelle-Jefferson** [comment]
Can you add Protoss Special Forces -2- from SyluX please? Thanks

**Reply by DarthVan** [reply]
ok, ill try soon.
